sound/hda/intel-sdw-acpi.c

```diff
@@ -79,13 +79,25 @@ sdw_intel_scan_controller(struct sdw_intel_acpi_info *info)
 	 *
 	 * We will check the hardware capabilities in the startup() step
 	 */
+	ret = fwnode_property_read_u32(acpi_fwnode_handle(adev),
+				       "mipi-sdw-manager-list", &list);
+	if (!ret) {
```

Comment:
I would think more of the usual pattern
```
  if (ret)
    goto try_count;

  count = ...;
  goto skip_count;
```
Or having this as a helper which will return it for you
```
ret = _get_list_of(..., &list);
if (ret)
 ...
```
In this case you can get rid of `list_ul`.